app/src/main/java/com/github/se/polyfit/ui/components/IngredientList.kt

```diff
@@ -0,0 +1,100 @@
+package com.github.se.polyfit.ui.components
+
+import android.util.Log
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ExperimentalLayoutApi
+import androidx.compose.foundation.layout.FlowRow
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.MoreVert
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import com.github.se.polyfit.ui.screen.Ingredient
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun IngredientList(
+    it: PaddingValues,
+    ingredients: List<Ingredient>,
+    potentialIngredients: List<Ingredient>,
+    onAddIngredient: (Int) -> Unit,
+) {
```

Comment:
```suggestion
    onAddIngredient: (Ingredient) -> Unit,
) {
```
This change would make the implementation of the meal ViewModel easier. It could also handle potential issues, for example, if the index of a list changes or something like that.

Reply:
This portion of code was directly moved from `IngredientScreen.kt`, which was written by @medwardson, to its own file as a component for readability. Maybe @medwardson can answer if this was an intentional decision

Reply:
It's just like this because I didn't have the ingredient type when creating the class. I think let's leave this as is for now and then it will be tackled once I'm doing the data integration